pkg/terminal/starbind/starlark.go

```diff
@@ -0,0 +1,286 @@
+		if len(args) != 2 {
+			return nil, decorateError(thread, fmt.Errorf("wrong number of arguments"))
+		}
+		path, ok := args[0].(starlark.String)
+		if !ok {
+			return nil, decorateError(thread, fmt.Errorf("first argument of write_file was not a string"))
+		}
+		err := ioutil.WriteFile(string(path), []byte(args[1].String()), 0640)
+		return starlark.None, decorateError(thread, err)
+	})
+	env.env[curScopeBuiltinName] = starlark.NewBuiltin(curScopeBuiltinName, func(_ *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		return env.interfaceToStarlarkValue(env.ctx.Scope()), nil
+	})
+	env.env[defaultLoadConfigBuiltinName] = starlark.NewBuiltin(defaultLoadConfigBuiltinName, func(_ *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		return env.interfaceToStarlarkValue(env.ctx.LoadConfig()), nil
+	})
+	return env
+}
+
+// Execute executes a script. Path is the name of the file to execute and
+// source is the source code to execute.
+// Source can be either a []byte, a string or a io.Reader. If source is nil
+// Execute will execute the file specified by 'path'.
+// After the file is executed if a function named mainFnName exists it will be called, passing args to it.
+func (env *Env) Execute(path string, source interface{}, mainFnName string, args []interface{}) (starlark.Value, error) {
+	defer func() {
+		err := recover()
+		if err == nil {
+			return
+		}
+		fmt.Printf("panic executing starlark script: %v\n", err)
+		for i := 0; ; i++ {
+			pc, file, line, ok := runtime.Caller(i)
+			if !ok {
+				break
+			}
+			fname := "<unknown>"
+			fn := runtime.FuncForPC(pc)
+			if fn != nil {
+				fname = fn.Name()
+			}
+			fmt.Printf("%s\n\tin %s:%d\n", fname, file, line)
+		}
+	}()
+
+	thread := env.newThread()
+	globals, err := starlark.ExecFile(thread, path, source, env.env)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	for name, val := range globals {
+		switch {
+		case strings.HasPrefix(name, commandPrefix):
+			err := env.createCallback(name, val)
+			if err != nil {
+				return starlark.None, err
+			}
+		case name[0] >= 'A' && name[0] <= 'Z':
+			env.env[name] = val
+		}
+	}
+
+	return env.callMain(thread, globals, mainFnName, args)
+}
+
+// Cancel cancels the execution of a currently running script or function.
+func (env *Env) Cancel() {
+	if env == nil {
+		return
+	}
+	env.contextMu.Lock()
+	if env.cancelfn != nil {
+		env.cancelfn()
+		env.cancelfn = nil
+	}
+	env.contextMu.Unlock()
+}
+
+func (env *Env) newThread() *starlark.Thread {
+	thread := &starlark.Thread{
+		Print: func(_ *starlark.Thread, msg string) { fmt.Println(msg) },
+	}
+	env.contextMu.Lock()
+	var ctx context.Context
+	ctx, env.cancelfn = context.WithCancel(context.Background())
+	env.contextMu.Unlock()
+	thread.SetLocal(dlvContextName, ctx)
+	return thread
+}
+
+func (env *Env) createCallback(name string, val starlark.Value) error {
```

Comment:
This should be changed for consistency with `RegisterCommand`.

Reply:
Done